src/transformers/modeling_utils.py

```diff
@@ -2096,7 +2096,8 @@ def _get_resized_embeddings(
         else:
             new_embeddings.weight.data[:n, :] = old_embeddings.weight.data[:n, :]
 
-        return new_embeddings
+        old_embeddings.weight.data = new_embeddings.weight.data
```

Comment:
Could we add a comment above to explain why we're doing this? Even just: 

```suggestion
        # c.f. https://github.com/huggingface/transformers/pull/31979
        old_embeddings.weight.data = new_embeddings.weight.data
```